File: app/src/main/java/com/example/bloodbank/ui/activity/BaseActivity.java
package com.example.bloodbank.ui.activity;

import androidx.appcompat.app.AppCompatActivity;

import com.example.bloodbank.ui.fragment.BaseFragment;

public class BaseActivity extends AppCompatActivity {
    public BaseFragment baseFragment;

    @Override
    public void onBackPressed() {
        baseFragment.onBack();
    }

    public void superBackPressed() {
        super.onBackPressed();
    }
}

File: app/src/main/java/com/example/bloodbank/ui/activity/BaseActivity.kt
package com.example.bloodbank.ui.activity

import androidx.appcompat.app.AppCompatActivity
import com.example.bloodbank.ui.fragment.BaseFragment

open class BaseActivity : AppCompatActivity() {
    lateinit var baseFragment: BaseFragment
    override fun onBackPressed() {
        baseFragment.onBack()
    }

    fun superBackPressed() {
        super.onBackPressed()
    }
}